vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogHeaderFooterIT.java

```diff
@@ -211,7 +203,25 @@ private void assertDialogNotContains(String text) {
     }
 
     private void verifyDialogOpened() {
-        waitForElementPresent(By.tagName("vaadin-dialog-overlay"));
+        waitForElementPresent(
+                By.cssSelector("vaadin-dialog-overlay div button"));
+    }
+
+    private void verifyContentRendered(String content) {
+        waitUntil(c -> {
+            try {
+                for (;;) {
```

Comment:
I don't think this loop is necessary, `waitUntil` already repeats the code below in an interval.

Reply:
done